src/NUnitEngine/nunit.engine.api/TestPackage.cs

```diff
@@ -86,32 +104,49 @@ public string Name
         public string FullName { get; private set; }
 
         /// <summary>
-        /// Gets an array of the test files contained in this package
+        /// Gets the list of SubPackages contained in this package
         /// </summary>
-        public IList<string> TestFiles
-        {
-            get { return _testFiles; }
-        }
+        public IList<TestPackage> SubPackages { get; private set; }
 
         /// <summary>
         /// Gets the settings dictionary for this package.
         /// </summary>
-        public IDictionary<string,object> Settings
-        {
-            get { return _settings; }
-        }
+        public IDictionary<string,object> Settings { get; private set; }
 
         #endregion
 
         #region Public Methods
 
         /// <summary>
-        /// Add a test file to the package.
+        /// Add a subproject to the package.
+        /// </summary>
+        /// <param name="subPackage">The subpackage to be added</param>
+        public void AddSubPackage(TestPackage subPackage)
+        {
+            SubPackages.Add(subPackage);
+
+            foreach (var key in Settings.Keys)
+                subPackage.Settings[key] = Settings[key];
+        }
+
+        /// <summary>
+        /// Add a setting to a package and all of its subpackages.
         /// </summary>
-        /// <param name="testFile">The test file to be added</param>
-        public void Add(string testFile)
+        /// <typeparam name="T">The Type of the setting value</typeparam>
+        /// <param name="name">The name of the setting</param>
+        /// <param name="value">The value of the setting</param>
+        /// <remarks>
+        /// Once a package is created, subpackages may have been created
+        /// as well. If you add a setting directly to the Settings dictionary
+        /// of the package, the subpackages are not updated. This method is
+        /// used when the settings are intended to be reflected to all the
+        /// subpackages under the package.
+        /// </remarks>
+        public void AddSetting<T>(string name, T value)
```

Comment:
Does this really need to be generic? The settings are collections of `object`, so why not just pass in an `object` for the `value`?